[release-2.5] Break ties between event timestamps

JustinKuli opened the pull request:
In some cases (it seems to be becoming more common), policy events may be batched together by policy controllers, and then sent in the same second. When this happens, status-sync could not accurately sort them. Since the controller will not re-send compliant events without other changes, this resulted in the policy and the template disagreeing on the compliance state.

Compliance events (by convention) suffix the name with a hexadecimal encoding of the time they were created, which is used here to break timestamp ties.

Cherrypicked from these upstream commits:
- df31d40aa58630b7757d177dced3212911143008
- 955ba8556a71740e94e13a61af5be707738a8b37
- be110ec7886a0972e6084ec63ee7783e146abe34
- 499ee07d8ba074e1f4469f18aff0d22c7e949474
- 55257b07b2960b58942a5997a06bc05da72f6b56

Signed-off-by: Justin Kulikauskas <jkulikau@redhat.com>

JustinKuli committed:
Break ties between event timestamps

In some cases (it seems to be becoming more common), policy events may
be batched together by policy controllers, and then sent in the same
second. When this happens, status-sync could not accurately sort them.
Since the controller will not re-send compliant events without other
changes, this resulted in the policy and the template disagreeing on
the compliance state.

Compliance events (by convention) suffix the name with a hexadecimal
encoding of the time they were created, which is used here to break
timestamp ties.

Cherrypicked from these upstream commits:
- df31d40aa58630b7757d177dced3212911143008
- 955ba8556a71740e94e13a61af5be707738a8b37
- be110ec7886a0972e6084ec63ee7783e146abe34
- 499ee07d8ba074e1f4469f18aff0d22c7e949474
- 55257b07b2960b58942a5997a06bc05da72f6b56

Signed-off-by: Justin Kulikauskas <jkulikau@redhat.com>

JustinKuli committed:
Use tesNamespace as clusterNamespaceOnHub

Signed-off-by: Justin Kulikauskas <jkulikau@redhat.com>

JustinKuli commented:
```
level=warning msg="[runner] Can't run linter goanalysis_metalinter: inspect: failed to load package : could not load export data: no export data for \"open-cluster-management.io/governance-policy-propagator/test/utils\""
level=error msg="Running error: 1 error occurred:\n\t* can't run linter goanalysis_metalinter: inspect: failed to load package : could not load export data: no export data for \"open-cluster-management.io/governance-policy-propagator/test/utils\"\n\n"
```
Had that locally too... hmmm

dhaiducek commented:
I almost forgot about this PR. Had you investigated the linter error?

JustinKuli committed:
Fix import of propagator

Signed-off-by: Justin Kulikauskas <jkulikau@redhat.com>

JustinKuli commented:
The new test was importing `open-cluster-management.io/governance-policy-propagator/test/utils` but all the other tests were importing `github.com/stolostron/governance-policy-propagator/test/utils`. Fixed.

dhaiducek commented:
Something odd is happening with the Kind `v1.18.15` deployment where it's not starting:
```
[kubelet-check] It seems like the kubelet isn't running or healthy.
[kubelet-check] The HTTP call equal to 'curl -sSL http://localhost:10248/healthz' failed with error: Get http://localhost:10248/healthz: dial tcp [::1]:10248: connect: connection refused.
```

JustinKuli committed:
Update kind node image

The 1.18 image seemed to stop working in github actions, and actually
the oldest version of OpenShift this needs to support uses k8s v1.21.

Signed-off-by: Justin Kulikauskas <jkulikau@redhat.com>

JustinKuli commented:
/override KinD tests (v1.18.15)

JustinKuli commented:
/override "KinD tests (v1.18.15)"

nelsonjean commented:
Related to https://github.com/stolostron/backlog/issues/26039
